### PR TITLE
verifier: expose Interface and hide implementation details

### PR DESCRIPTION
**Pull request by soltysh**
The code currently wrongly uses implementation types rather than interfaces, this code fixes that by exposing an interface and ensuring all the types implement it properly. 

/assign @wking @jottofar

### CODE REVIEW

File: pkg/verify/verify.go [hunk 1]
```diff
@@ -48,10 +63,10 @@ const maxSignatureSearch = 10
 // validReleaseDigest is a verification rule to filter clearly invalid digests.
 var validReleaseDigest = regexp.MustCompile(`^[a-zA-Z0-9:]+$`)
 
-// ReleaseVerifier implements a signature intersection operation on a provided release
+// releaseVerifier implements a signature intersection operation on a provided release
 // digest - all verifiers must have at least one valid signature attesting the release
 // digest. If any failure occurs the caller should assume the content is unverified.
-type ReleaseVerifier struct {
+type releaseVerifier struct {
```
Comment (jottofar) [inline pkg/verify/verify.go:69]: CVO, which will be reusing this package as well, needs access to field Store. Prior to any of my changes or refactoring to library-go ReleaseVerifier was also exported.

Comment (soltysh) [reply pkg/verify/verify.go:69]: If we want to re-use the interface, I'd suggest adding `Store` method to interface, directly accessing struct fields won't help here. And the current approach is causing nil-pointer exception when used in oc.

File: pkg/verify/verify.go [hunk 2]
```diff
@@ -200,8 +222,13 @@ func (v *ReleaseVerifier) Signatures() map[string][][]byte {
 	return copied
 }
 
+// AddStore adds additional stores for signature verification.
+func (v *releaseVerifier) AddStore(additionalStore store.Store) {
+	v.store = &serial.Store{Stores: []store.Store{additionalStore, v.store}}
```
Comment (wking) [inline pkg/verify/verify.go:227]: do we always want to add with a serial prepend?  I think we might want a getter/setter method pair so folks can pick the combination on the caller-side.

Comment (jottofar) [reply pkg/verify/verify.go:227]: This was just lifted exactly from CVO's current setting of 'verifier.Store` but also fine to go with the more flexible approach as well.

Comment (soltysh) [reply pkg/verify/verify.go:227]: That's the only usage I've noticed. I'd suggest you guys have a closer look at that package implementation and reconsider the overall shape, imo. For now I'd go with as is, and I'd definitely discourage getters & setters.

Comment (jottofar) [reply pkg/verify/verify.go:227]: As discussed with @wking, this will satisfy CVO's needs now and we can change it later if necessary.

**Pull request merged**